### PR TITLE
fix: focus indicator clipping.

### PR DESCRIPTION
**Pull request by richburdon**
Is there a more systematic fix?

Screenshot 2024-11-30 at 11 39 45 PM

### CODE REVIEW

File: packages/sdk/shell/src/panels/IdentityPanel/steps/IdentityActionChooser.tsx [hunk 1]
```diff
@@ -63,7 +63,8 @@ export const IdentityActionChooserImpl = ({
     (device) => device.profile?.type === DeviceType.AGENT_MANAGED && device.profile?.os?.toUpperCase() === 'EDGE',
   );
   return (
-    <div role='none' className='bs-40 grow overflow-y-auto overflow-x-hidden'>
+    // TODO(burdon): Create box container to handle padding for focus indicators so they are not clipped.
+    <div role='none' className='bs-40 grow overflow-y-auto px-1'>
```
Comment (thure) [inline packages/sdk/shell/src/panels/IdentityPanel/steps/IdentityActionChooser.tsx:67]:
```suggestion
    <div role='none' className='bs-40 grow overflow-y-auto pli-1'>
```

**Review by thure — APPROVED** [review]

**Comment by thure** [comment]
The only silver bullet is to inset all focus rings.

**Comment by richburdon** [comment]
> The only silver bullet is to inset all focus rings.
> 
> Another strategy might be to enforce padding on overflow values other than visible, though that could cause other issues.

Wouldn't the following work?

- theme adds sufficient padding to all container components that contain focusable elements
- i.e., Toolbar, List, Tree

**Comment by thure** [comment]
> - i.e., Toolbar, List, Tree

It would certainly be a bug if any of those components didn't accommodate focus rings themselves, but the element you added spacing to here isn't an intermediate component like these.

I don't think there is a cross-cutting solution to this that wouldn't make debugging this more difficult than simply adjusting spacing where we neglected to as in this PR.

**Pull request merged**